### PR TITLE
Cleanup for the Pointer changes.

### PR DESCRIPTION
**Pull request by Hixie**
TBR @abarth 
cc @vlidholt for game engine changes

:warning: the flutter_sprites changes are untested and unanalyzed, since I couldn't figure out how to do either. See #759 and #761 .

### CODE REVIEW

File: packages/flutter_sprites/lib/src/node.dart [hunk 1]
```diff
@@ -769,10 +769,10 @@ class Node {
   ///       }
   ///
   ///       bool handleEvent(SpriteBoxEvent event) {
-  ///         if (event.type == 'pointerdown) {
+  ///         if (event.type == PointerDownEvent) {
```
Comment (abarth) [inline packages/flutter_sprites/lib/src/node.dart]: `s/==/is/`

File: packages/flutter_sprites/lib/src/node.dart [hunk 2]
```diff
@@ -769,10 +769,10 @@ class Node {
...
   ///           opacity = 0.5;
   ///         }
-  ///         else if (event.type == 'pointerup') {
+  ///         else if (event.type == PointerUpEvent) {
```
Comment (abarth) [inline packages/flutter_sprites/lib/src/node.dart]: And lose the ".type"

**Comment by abarth** [comment]
LGTM

**Comment by Hixie** [comment]
cc @krisgiesing since a lot of this was your code

File: packages/flutter_sprites/lib/src/node.dart (hunk 1 above)
Comment (krisgiesing) [inline packages/flutter_sprites/lib/src/node.dart]: I think "event.type == PointerDownEvent" should be "event is PointerDownEvent" here?

File: packages/flutter_sprites/lib/src/node.dart (hunk 2 above)
Comment (krisgiesing) [inline packages/flutter_sprites/lib/src/node.dart]: Ditto above

File: packages/flutter_sprites/lib/src/virtual_joystick.dart [hunk 3]
```diff
@@ -31,18 +31,18 @@ class VirtualJoystick extends NodeWithSize {
   Paint _paintControl;
 
   bool handleEvent(SpriteBoxEvent event) {
-    if (event.type == "pointerdown") {
+    if (event.type == PointerDownEvent) {
```
Comment (krisgiesing) [inline packages/flutter_sprites/lib/src/virtual_joystick.dart]: "event is PointerDownEvent"?  I'm not entirely sure why this isn't giving an analyzer warning.

**Comment by krisgiesing** [comment]
I think I'm confused about "event is ..." vs. "event.type == ...". Are they both valid?

**Comment by Hixie** [comment]
@krisgiesing The events in the flutter_sprites library are not the same class. See https://github.com/flutter/flutter/issues/761 for details.

**Comment by krisgiesing** [comment]
Ah, I see now, thanks.

**Pull request merged**